examples/Components/animationloop/MechanicalMatrixMapperExample.pyscn

```diff
@@ -0,0 +1,50 @@
+import Sofa
+import random	
+
+def createScene(rootNode):
+
+        rootNode.createObject('RequiredPlugin', pluginName='ModelOrderReduction')
+        rootNode.createObject('VisualStyle', displayFlags='showCollisionModels hideMappings showForceFields')
+
+        meshDivision = rootNode.createChild('meshDivision')
+        meshDivision.createObject('RegularGridTopology', name='topology', n="2 2 4" , min="-1 -1 -2", max="1 1 2")
+        meshOfStructure = meshDivision.createObject('Mesh',name='beamMesh', src="@topology")
+        meshDivision.createObject('MechanicalObject', template='Vec3d')
+        boxFixed= meshDivision.createObject('BoxROI',template="Vec3d", name='box_roi_fix',box='-1 -1 -2.05  1 1 -1.5' ,drawBoxes='1', drawPoints='1' , position="@beamMesh.position" )
+        boxDeformable= meshDivision.createObject('BoxROI',template="Vec3d", name='box_roi_deformable',box='-1 -1 -2.05  1   1 1.5',drawBoxes='1', drawPoints='1', position="@beamMesh.position" )
+        boxRigid= meshDivision.createObject('BoxROI',template="Vec3d", name='box_roi_rigid',box='-1 -1 1.500001 1 1 2.05',drawBoxes='1', drawPoints='1', position="@beamMesh.position")
+
+        SolverNode= rootNode.createChild('SolverNode')
+        SolverNode.createObject('EulerImplicit',verbose='false')
+        SolverNode.createObject('SparseLDLSolver', name="ldlsolveur")
+        #SolverNode.createObject('MechanicalMatrixMapper', template='Vec3d,Rigid', object1='@./deformablePartNode/beamPart1Mech', object2='@./RigidNode/rigid1', nodeToParse='@./deformablePartNode/FEMNode')
+
+
+#####   Deformable Part of the BEAM (Main Body)
+        deformablePartNode= SolverNode.createChild('deformablePartNode')              
+        deformablePartNode.createObject('MechanicalObject', template='Vec3d',name='beamPart1Mech', position="@../../meshDivision/box_roi_deformable.pointsInROI")
+
+#####   Rigid Part of the BEAM (Top)
+        RigidNode= SolverNode.createChild('RigidNode')
+        RigidNode.createObject("MechanicalObject",template="Rigid",name="rigid1", position="0 2 0 0 0 0 1")
+        RigidifiedNode= RigidNode.createChild('RigidifiedNode')
+
+        RigidifiedNode.createObject("MechanicalObject",name="rigidMecha",template="Vec3d", position="@../../../meshDivision/box_roi_rigid.pointsInROI")
+        RigidifiedNode.createObject("RigidMapping",globalToLocalCoords="true")  
+
+#####   COMBINED BEAM 	        
+        FEMNode= deformablePartNode.createChild('FEMNode')
+        RigidifiedNode.addChild(FEMNode)
+
+        FEMNode.createObject('Mesh',name='meshInput',src="@../../../../meshDivision/beamMesh")                     
```

Comment:
You should replace link made by string like "@../../../../meshDivision/beamMesh" with automatically generated link from the target object eg: "@"+meshOfStructure.getPathName() 

It is more robust against change of location & naming of the targeted components.